autoprotocol/builders.py

```diff
@@ -1738,6 +1745,154 @@ def blowout(self, volume, initial_z, flowrate=None):
+        TypeError
+            liquid is not of str
+        ValueError
+            liquid_class is not a valid value
+        ValueError
+            liquid_class did not resolve to a mode
+        ValueError
+            multiple liquid_class exists in one LiquidHandle
+        """
+        # dict for default dispense mode for each liquid_class
+        liquid_class_to_dispense_modes = {
+            "air": "air_displacement",
+            "default": "air_displacement",
+            "viscous": "positive_displacement"
+        }
+        if mode:
+            if not isinstance(mode, str):
+                raise TypeError(
+                    "mode: {} must be in str".format(mode)
+                )
+            if mode not in self.dispense_modes:
+                raise ValueError(
+                    "mode: {} must be one of the valid modes: {}"
+                    "".format(mode, self.dispense_modes)
+                )
+        else:
+            # get liquid_classes from transport input
+            liquid_classes = set([transport["mode_params"]["liquid_class"]
+                                  for transport in transports])
+            # remove automatically added 'air' class from blowout, etc.
+            non_air_classes = [liq for liq in liquid_classes if liq != "air"]
+            # if liquid classes only contain "air" type, mode is returned.
+            if not non_air_classes:
+                return liquid_class_to_dispense_modes["air"]
+
+            modes = []
+            # get mode for each liquid_class if not specified already from the
+            # user input.
+            for liquid in non_air_classes:
+                # resolve mode for different liquid classes (except for None).
+                if liquid:
+                    if not isinstance(liquid, str):
+                        raise TypeError(
+                            "liquid: {} is not of type str".format(liquid)
+                        )
+                    if liquid not in self.liquid_classes:
+                        raise ValueError(
+                            "liquid_class: {} must be one of the valid"
+                            "classes: {} ".format(liquid, self.liquid_classes)
+                        )
+                    if liquid not in liquid_class_to_dispense_modes.keys():
+                        raise ValueError(
+                            "modes: {} did not resolve accordingly. If there "
+                            "is a new liquid_class, make sure dictionary: {}"
+                            "is updated.".format(
+                                set(modes), liquid_class_to_dispense_modes
+                            )
+                        )
+                    modes.append(liquid_class_to_dispense_modes[liquid])
+
+            # if liquid class only contains "air" and/or None, default to
+            # "air_displacement".
+            if not modes:
+                modes.append("air_displacement")
```

Comment:
Probably should use the dictionary here `liquid_class_to_dispense_modes["air"].`

Reply:
oops. missed this one. updating!